docs/dev/release-process.md

```diff
@@ -0,0 +1,26 @@
+# The Release Process
+
+This project currently follows a [Continuous Delivery](https://en.wikipedia.org/wiki/Continuous_delivery) process, but it's gradually moving toward [Continuous Deployment](https://en.wikipedia.org/wiki/Continuous_deployment).
+
+Whenever a commit is merged to this repository's `main` branch, the deployment pipeline kicks in, deploying the changeset to the [`stage` environment](../firefox.md#stage).
+After the deployment is complete, accessing the [`__version__` endpoint](https://stage.merino.nonprod.cloudops.mozgcp.net/__version__) will show the commit hash of the deployed version, which will match to the one of the latest commit on the `main` branch.
```

Comment:
The following may not always be the case:

> which will match to the one of the latest commit on the `main` branch

Given that the updated CI workflow will skip the deployment if it finds a `[do not deploy]` text in the commit message, the latest commit on the `main` branch may not be deployed to `STAGE`. Also while the deployment is in progress, the `/__version__` may be handled by a node that isn't running the latest version (but that's an edge case).

Reply:
The sentence starts with `After the deployment is complete`, so I'm assuming that deployment happened in the rest of the sentence. Happy to accept suggestions on making this clearer if it is uncear, though.

Good point on the version part, I can add that it will "eventually" match given that the endpoint could be served by an older version still living.